### DISCUSSION TITLE
variables in include path setting

### DISCUSSION THREAD

**Discussion by hmoffatt** [microsoft/vscode-cpptools#10976] [Q&A]
Should VSCode variables work in the include path setting?

I'm trying to set `C_Cpp.default.includePath` to include `${userHome}/some/path/include/**`, but it's not working. If I write `/home/hamish/...` instead then it works.

**Comment by sean-mcmanus** [comment]
Yeah, it works for me. Can you try remove the `"**"` temporarily and then run the C/C++: Log Diagnostics command and look for that include path in the logging? Using `"**"` will cause the include paths to be removed if they aren't actually referenced. Or can you provide any more repro info?

**Comment by hmoffatt** [comment]
In the extension settings I added
`${userHome}/dev/Qt5.15.12/5.15.12/gcc_64/include/QtCore` and one other.

In `c_cpp_properties.json` I have
```
            "includePath": [
                "${default}",
                "${workspaceFolder}/**"
            ],
```

The log diagnostics command produces:

```
-------- Diagnostics - 24/05/2023, 18:31:05
Version: 1.15.4
Current Configuration:
{
    "name": "Linux",
    "includePath": [
        "/home/hamish/dev/trunk/src/${userHome}/dev/Qt5.15.12/5.15.12/gcc_64/include/QtCore",
        "/home/hamish/dev/trunk/src/${userHome}/dev/Qt5.15.12/5.15.12/gcc_64/include",
        "/home/hamish/dev/trunk/src/**"
    ],
    "defines": [],
    "compilerPath": "/usr/bin/clang",
    "cStandard": "c17",
    "cppStandard": "c++14",
    "intelliSenseMode": "linux-clang-x64",
    "compilerPathIsExplicit": true,
    "cStandardIsExplicit": true,
    "cppStandardIsExplicit": true,
    "intelliSenseModeIsExplicit": true,
    "compilerPathInCppPropertiesJson": "/usr/bin/clang",
    "mergeConfigurations": false,
    "browse": {
        "path": [
            "${userHome}/dev/Qt5.15.12/5.15.12/gcc_64/include/QtCore",
            "${userHome}/dev/Qt5.15.12/5.15.12/gcc_64/include",
            "${workspaceFolder}/**"
        ],
        "limitSymbolsToIncludedHeaders": true
    }
}
```

**Comment by hmoffatt** [comment]
It works if I use `${env:HOME}` instead of `${userHome}`.

Glancing at the source, I think only `workspaceFolder`, `workspaceFolderBasename` and `env:` may be supported, none of the other variables documented at https://code.visualstudio.com/docs/editor/variables-reference

**Reply by sean-mcmanus** [reply]
Yeah, we don't support all the variables defined in https://code.visualstudio.com/docs/editor/variables-reference . Those are not environment variables. There's a previous request at https://github.com/microsoft/vscode-cpptools/issues/10350 . `${HOME}` works for me though, because it's an environment variable.

Oh, looks like I misread your original reference to "VSCode variables".
